src/onegov/org/forms/settings.py

```diff
@@ -1132,3 +1133,39 @@ class EventSettingsForm(Form):
         description=_('Enables website visitors to submit their own events'),
         default=True
     )
+
+
+def generate_timespans() -> (
+    tuple[tuple[str | datetime.timedelta, str], ...]
+):
+    return (
+        ('disabled', _('Disabled')),
+        (datetime.timedelta(days=180), _('6 months')),
+        (datetime.timedelta(days=365), _('1 year')),
+        (datetime.timedelta(days=365 * 2), _('2 years')),
+        (datetime.timedelta(days=365 * 3), _('3 years')),
+    )
+
+
+class DataRetentionPolicyForm(Form):
+
+    auto_archive_timespan = RadioField(
+        label=_('Duration from opening a ticket to its automatic archival'),
+        validators=[InputRequired()],
+        default='disabled',
+        choices=generate_timespans()
+    )
+
+    auto_delete_timespan = RadioField(
+        label=_('Duration from archived state until deleted automatically'),
+        validators=[InputRequired()],
+        default='disabled',
+        choices=generate_timespans()
+    )
+
+    def process_obj(self, model):
+        super().process_obj(model)
```

Comment:
You don't actually need to call `super().process_obj(model)` here, since you are manually setting the only fields in the form.

But by adressing the above comment you may be able to get rid of this entirely anyways.